src/java/org/apache/cassandra/metrics/ClientMetrics.java

```diff
@@ -51,20 +84,62 @@ public final class ClientMetrics
     private Meter protocolException;
     private Meter unknownException;
 
+    private static String AUTH_SUCCESS = "AuthSuccess";
+
+    private static String AUTH_FAILURE = "AuthFailure";
+
+    private static String CONNECTED_NATIVE_CLIENTS = "ConnectedNativeClients";
+
     private ClientMetrics()
     {
     }
 
+    /**
+     * @deprecated by {@link #markAuthSuccess(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthSuccess()
     {
-        authSuccess.mark();
+        markAuthSuccess(null);
+    }
+
+    public void markAuthSuccess(final String mode)
+    {
+        markAuthMeter(authSuccess, AUTH_SUCCESS, mode);
     }
 
+    /**
+     * @deprecated by {@link #markAuthFailure(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthFailure()
     {
         authFailure.mark();
     }
 
+    public void markAuthFailure(final String mode)
+    {
+        markAuthMeter(authFailure, AUTH_FAILURE, mode);
+    }
+
+    private void markAuthMeter(final Meter meter, final String meterPrefix, final String mode)
+    {
+        meter.mark();
+
+        if (mode != null)
```

Comment:
would not be it easier to read if we `return` when `mode == null`? Then you would not put this whole logic into `if` and a reader does not need to hold the context of "ok, mode is not null here" in their head. Just return / break the execution flow as soon as possible.